### PR TITLE
Demo API: Restrict import of `@IsOptional()` validator

### PR DESCRIPTION
**Pull request by johnnyomair**
## Description

We introduced `@IsNullable()` and `@IsUndefinable()` [a while ago](https://github.com/vivid-planet/comet/pull/1112), but never enforced it, since updating applications is cumbersome. This change restricts the import of `@IsOptional()` in Demo (and later Starter) to ensure that new applications use the correct decorators.

### CODE REVIEW

File: .changeset/gorgeous-rocks-shave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": patch
+---
+
+API Generator: Use `@IsUndefinable()` instead of `@IsOptional()`
```
Comment (johnnyomair) [inline .changeset/gorgeous-rocks-shave.md:5]: Kinda sad that even the API Generator didn't use it 😅

File: .env [hunk 2]
```diff
@@ -13,7 +13,6 @@ POSTGRESQL_USER=postgres
 POSTGRESQL_PWD=dml2aWQK
 POSTGRESQL_PWD_DECODED=vivid
 POSTGRESQL_DB=main
-POSTGRESQL_USE_SSL=false
```
Comment (johnnyomair) [inline .env:16]: Remove this to test if the usage [here](https://github.com/vivid-planet/comet/pull/3531/files#diff-e11ed660bb5b0c5587616338c6a60932e42ff9f53b9154ea25b9ec8c410cba1cR14) is correct.

File: demo/api/src/news/blocks/news-detail.block.ts [hunk 3]
```diff
@@ -24,7 +25,7 @@ class NewsDetailBlockData extends BlockData {
 class NewsDetailBlockInput extends BlockInput {
     @BlockField({ nullable: true })
     @IsUUID()
-    @IsOptional()
+    @IsUndefinable()
     id?: string;
```
Comment (nsams) [inline demo/api/src/news/blocks/news-detail.block.ts:29]: idn't nullable here correct?

(we don't support partial update for block input)

Comment (johnnyomair) [reply demo/api/src/news/blocks/news-detail.block.ts:29]: Block fields are usually `undefined`, not `null`. Therefore I chose `@IsUndefinable()`. I've also tested it with an empty input, and it works.

> (we don't support partial update for block input)

Yes, here setting it to undefined makes it undefined, which is correct since it's "just" a JSON field.

**Review by kaufmo — APPROVED** [review]

**Review by dkarnutsch — APPROVED** [review]

File: demo/api/.eslintrc.json [hunk 4]
```diff
@@ -1,4 +1,18 @@
 {
     "extends": "@comet/eslint-config/nestjs",
-    "ignorePatterns": ["src/db/migrations/**", "dist/**"]
+    "ignorePatterns": ["src/db/migrations/**", "dist/**"],
+    "rules": {
+        "no-restricted-imports": [
```
Comment (manuelblum) [inline demo/api/.eslintrc.json:5]: should we also restrict it in @comet/eslint-config to bring it to the projects?

Comment (nsams) [reply demo/api/.eslintrc.json:5]: we decided against this, as it is difficult to update existing applications

Comment (manuelblum) [reply demo/api/.eslintrc.json:5]: I can't remember this.

How do we intend to handle new Projects then?

In my opinion we should do one of the following to have this rule in NEW projects:

- **Version 1:** add no-restricted-imports to @comet/eslint-config/nestjs.js
     -  deactivate in existing projects (manually or with codemod)
- **Version 2:** create a new version of  eslint-config/nestjs.js (e.g. nestjsv2.js)

I would tend to Version 1.

Comment (johnnyomair) [reply demo/api/.eslintrc.json:5]: > How do we intend to handle new Projects then?

New projects will be created with the Starter, so they will have the import restriction

> deactivate in existing projects (manually or with codemod)

The problem is that we can't disable no-restricted-import for a single import. Therefore disabling the rule would disable all import restrictions.

Comment (manuelblum) [reply demo/api/.eslintrc.json:5]: it's not easy and maybe not straight forward, but it could work like this PoC targetting on the next branch: https://github.com/vivid-planet/comet/pull/3577
